pkg/api/aim/query/query_test.go

```diff
@@ -32,36 +34,132 @@ func (s *QueryTestSuite) SetupTest() {
+			expectedVars: []interface{}{"%run%", models.LifecycleStageDeleted},
 		},
 		{
 			name:         "TestRunNameWithStartWithFunction",
 			query:        `(run.name.startswith('run'))`,
-			expectedSQL:  `SELECT * FROM "runs" WHERE ("runs"."name" LIKE 'run%' AND "runs"."lifecycle_stage" <> $1) ORDER BY "runs"."run_uuid" LIMIT 1`,
-			expectedVars: []interface{}{models.LifecycleStageDeleted},
+			dialector:    postgres.Dialector{}.Name(),
+			expectedSQL:  `SELECT * FROM "runs" WHERE ("runs"."name" LIKE $1 AND "runs"."lifecycle_stage" <> $2) ORDER BY "runs"."run_uuid" LIMIT 1`,
+			expectedVars: []interface{}{"run%", models.LifecycleStageDeleted},
 		},
 		{
 			name:         "TestRunNameWithEndWithFunction",
 			query:        `(run.name.endswith('run'))`,
-			expectedSQL:  `SELECT * FROM "runs" WHERE ("runs"."name" LIKE '%run' AND "runs"."lifecycle_stage" <> $1) ORDER BY "runs"."run_uuid" LIMIT 1`,
-			expectedVars: []interface{}{models.LifecycleStageDeleted},
+			dialector:    postgres.Dialector{}.Name(),
+			expectedSQL:  `SELECT * FROM "runs" WHERE ("runs"."name" LIKE $1 AND "runs"."lifecycle_stage" <> $2) ORDER BY "runs"."run_uuid" LIMIT 1`,
+			expectedVars: []interface{}{"%run", models.LifecycleStageDeleted},
+		},
+		{
+			name:         "TestRunNameWithRegexpMatchFunction",
+			query:        `(re.match('run', run.name))`,
+			dialector:    postgres.Dialector{}.Name(),
+			expectedSQL:  `SELECT * FROM "runs" WHERE ("runs"."name" ~ $1 AND "runs"."lifecycle_stage" <> $2) ORDER BY "runs"."run_uuid" LIMIT 1`,
+			expectedVars: []interface{}{"^run", models.LifecycleStageDeleted},
+		},
+		{
+			name:         "TestRunNameWithRegexpSearchFunction",
+			query:        `(re.search('run', run.name))`,
+			dialector:    postgres.Dialector{}.Name(),
+			expectedSQL:  `SELECT * FROM "runs" WHERE ("runs"."name" ~ $1 AND "runs"."lifecycle_stage" <> $2) ORDER BY "runs"."run_uuid" LIMIT 1`,
+			expectedVars: []interface{}{"run", models.LifecycleStageDeleted},
+		},
+	}
+
+	for _, tt := range tests {
+		s.T().Run(tt.name, func(T *testing.T) {
+			pq := QueryParser{
+				Default: DefaultExpression{
+					Contains:   "run.archived",
+					Expression: "not run.archived",
+				},
+				Tables: map[string]string{
+					"runs":        "runs",
+					"experiments": "Experiment",
+				},
+				Dialector: tt.dialector,
+			}
+			parsedQuery, err := pq.Parse(tt.query)
+			assert.Nil(s.T(), err)
+			result := parsedQuery.Filter(
+				s.db.Session(&gorm.Session{DryRun: true}).Model(models.Run{}),
+			).First(&models.Run{})
+			assert.Nil(s.T(), result.Error)
+			assert.Equal(s.T(), tt.expectedSQL, result.Statement.SQL.String())
+			assert.Equal(s.T(), tt.expectedVars, result.Statement.Vars)
+		})
+	}
+}
+
+func (s *QueryTestSuite) TestSqliteDialector_Ok() {
+	tests := []struct {
+		name         string
+		query        string
+		dialector    string
```

Comment:
since this test is entirely for sqlitedialector, this could be defaulted

Reply:
yep, removed.